source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/ACRChatWindow.mm

```diff
@@ -146,8 +144,61 @@ - (void)renderCards:(NSString *)card
                            widthConstraint:adaptiveCardsWidth
                                   delegate:self.adaptiveCardsDelegates];
     }
+    return renderResult.view;
+}
 
-    [adaptiveCardsViews addObject:renderResult.view];
+- (void)renderCards:(NSString *)card
+{
+    [adaptiveCardsViews addObject:[self renderCard:card]];
+}
+
+// empty the cached adaptive cards views
+// only visible adaptive card view will be drawn initially, and will cause to render adaptive cards when the cards in the invisible rows become visible.
+- (void)prepareForRedraw
+{
+    if (adaptiveCardsViews) {
+        for (NSInteger i = 0; i < adaptiveCardsViews.count; i++) {
+            adaptiveCardsViews[i] = [NSNull null];
+        }
+    }
+}
+@end
+
+// configure the content cell
+@implementation ACRChatWindowCell {
+    NSArray<NSLayoutConstraint *> *_contentViewConstratins;
+}
+
+- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
+{
+    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
+    if (self) {
+        _contentViewConstratins = [[NSMutableArray alloc] init];
+    }
+    return self;
+}
+
+- (void)setAdaptiveCardView:(ACRView *)adaptiveCardView
+{
+    _adaptiveCardView = adaptiveCardView;
+    if (self.adaptiveCardView) {
+        if (self.contentView.subviews && self.contentView.subviews.count) {
+            [self.contentView.subviews[0] removeFromSuperview];
+        }
+        [self.contentView addSubview:self.adaptiveCardView];
+        [self updateLayoutConstraints];
+    }
+}
+
+- (void)updateLayoutConstraints
+{
+    _contentViewConstratins = @[
+        [self.adaptiveCardView.centerXAnchor constraintEqualToAnchor:self.contentView.centerXAnchor],
+        [self.adaptiveCardView.centerYAnchor constraintEqualToAnchor:self.contentView.centerYAnchor],
+        [self.contentView.heightAnchor constraintEqualToAnchor:self.adaptiveCardView.heightAnchor],
+        [self.contentView.widthAnchor constraintEqualToAnchor:self.adaptiveCardView.widthAnchor]
+    ];
+    [NSLayoutConstraint activateConstraints:_contentViewConstratins];
```

Comment:
thank you for taking on this task, you may want to add "Fixes #6207" to the description